lib/modules/ai_bot/commands/image_command.rb

```diff
@@ -53,55 +65,78 @@ def process(prompt:)
       engine = SiteSetting.ai_stability_engine
       api_url = SiteSetting.ai_stability_api_url
 
-      # API is flaky, so try a few times
-      3.times do
-        begin
-          thread =
-            Thread.new do
-              begin
-                results =
-                  DiscourseAi::Inference::StabilityGenerator.perform!(
-                    prompt,
-                    engine: engine,
-                    api_key: api_key,
-                    api_url: api_url,
-                  )
-              rescue => e
-                Rails.logger.warn("Failed to generate image for prompt #{prompt}: #{e}")
-              end
+      i = 0
+      threads =
+        prompts.map do |prompt|
+          seed = seeds ? seeds[i] : nil
+          i += 1
+          Thread.new(seed, prompt) do |inner_seed, inner_prompt|
```

Comment:
I didn't now you could pass args to `Thread` like this, nice

Reply:
yeah, if you dont do that you can get some terrible concurrency bugs when stuff changes underneath you.